src/main/java/cat/udl/eps/softarch/linkapp/domain/UserRole.java

```diff
@@ -0,0 +1,34 @@
+package cat.udl.eps.softarch.linkapp.domain;
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+
+import javax.persistence.*;
+
+
+@Entity
+@Table(name = "UserRole",
+        uniqueConstraints = {
+                @UniqueConstraint(
+                        columnNames = {
+                                "user_id",
+                                "group_id"
+                        }
+                )
+        }
+)
+@Data
+@EqualsAndHashCode(callSuper = true)
+public class UserRole extends UriEntity<UserRoleKey> {
+
+    @EmbeddedId
```

Comment:
Even though this is the way to use foreign keys as identifiers, this implementation will cause trouble when creating this class through an http request. If you want, I can accept it for now and you can try to use it this way. The other option would be to define an auto-generated identifier and two relations with the entities.

Reply:
Looking online I can't find a problem with this way of defining the constraints. It should create a constraint on the DB and prevent repeated entries. What problem could this bring?